tests/spread/smoketests/metafiles/combine_yaml/task.yaml

```diff
@@ -0,0 +1,125 @@
+summary: >
+  create a charm that use charmcraft.yaml, 
+  metadata.yaml, config.yaml, actions.yaml
+
+include: 
+  - tests/
+
+prepare: |
+  tests.pkgs install unzip
+  charmcraft init --project-dir=charm
+  rm -f charm/metadata.yaml
+  rm -f charm/config.yaml
+
+  cat <<- EOF > charm/charmcraft.yaml
+  type: charm
+
+  bases:
+    - build-on:
+      - name: ubuntu
+        channel: "22.04"
+    run-on:
+      - name: ubuntu
+        channel: "22.04"
+
+  EOF
+
+  cat <<- EOF > charm/metadata.yaml
+  name: test-charm-name
+  summary: test-summary
+  description: test-description
+
+  title: test-title
+
+  links:
+    issues: https://example.com/issues
+    contact:
+      - https://example.com/contact
+      - contact@example.com
+      - "IRC #example"
+    documentation: https://example.com/docs
+    source:
+      - https://example.com/source
+      - https://example.com/source2
+      - https://example.com/source3
+    website:
+      - https://example.com/
+
+  #### TEST-COPY ####
+  EOF
+
+  cat <<- EOF > charm/actions.yaml
+  pause:
+    description: Pause the database.
+  resume:
+    description: Resume a paused database.
+  snapshot:
+    description: Take a snapshot of the database.
+    params:
+      filename:
+        type: string
+        description: The name of the snapshot file.
+      compression:
+        type: object
+        description: The type of compression to use.
+        properties:
+          kind:
+            type: string
+            enum: [gzip, bzip2, xz]
+          quality:
+            description: Compression quality
+            type: integer
+            minimum: 0
+            maximum: 9
+    required: [filename]
+    additionalProperties: false
+  #### TEST-COPY ####
+
+  EOF
+
+  cat <<- EOF > charm/config.yaml
+  options:
+    test-int:
+      default: 123
+      description: test-1
+      type: int
+    test-string:
+      description: test-2
+      type: string
+    test-float:
+      default: 1.23
+      type: float
+    test-bool:
+      default: true
+      type: boolean
+  #### TEST-COPY ####
+
+  EOF
+
+restore: |
+  pushd charm
+  charmcraft clean
+  popd
+  
+  rm -rf charm
+
+execute: |
+  cd charm
+  charmcraft pack --verbose
+  test -f charm*.charm
+
+  unzip -l charm*.charm | MATCH "actions.yaml"
+  unzip -l charm*.charm | MATCH "config.yaml"
+  unzip -l charm*.charm | MATCH "metadata.yaml"
+
+  unzip -p actions.yaml charm*.charm | MATCH "description: The type of compression to use."
+  unzip -p actions.yaml charm*.charm | MATCH "#### TEST-COPY ####"
```

Comment:
This is really clever!

![](https://teeturtle-s3-web.s3.amazonaws.com/accounts/1/products/1986199881350/Seal-of-Approval_800x800_SEPS-1000x1000.jpg)